packages/web-components/@sfx/sayt/test/sayt.test.ts

```diff
@@ -65,6 +96,210 @@ describe('Sayt Component', () => {
+      const event = { detail: { searchbox } };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should return true if event does not provide a searchbox ID', () => {
+      sayt.searchbox = 'some-searchbox-id';
+      const event = { detail: {} };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should return false if event provides the wrong searchbox ID', () => {
+      sayt.searchbox = 'correct-searchbox-id';
+      const event = { detail: { searchbox: 'wrong-searchbox-id' } };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.false;
+    });
+
+    it('should return true if event specifies a searchbox but SAYT has none specified', () => {
+      sayt.searchbox = undefined;
+      const event = { detail: { searchbox: 'some-searchbox-id' } };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should not error if passed an event without a detail attribute', () => {
+      const event = {};
+      const callback = () => sayt.isCorrectSayt(event);
+
+      expect(callback).to.not.throw();
+    });
+  });
+
+  describe('processClick()', () => {
+    let node: any = 'some-node';
+    let event: any;
+
+    beforeEach(() => {
+      event = { target: node };
+    });
+
+    it('should hide SAYT if the event target is nowhere relevant', () => {
+      stub(sayt, 'contains').returns(false);
+      stub(sayt, 'nodeInSearchBar').returns(false);
+      stub(sayt, 'hideSayt');
+
+      sayt.processClick(event);
+
+      expect(sayt.contains).to.be.calledWith(node);
+      expect(sayt.hideSayt).to.be.called;
+    });
+
+    it('should not SAYT if the event target is contained by SAYT', () => {
+      stub(sayt, 'contains').returns(true);
+      stub(sayt, 'hideSayt');
+
+      sayt.processClick(event);
+
+      expect(sayt.contains).to.be.calledWith(node);
+      expect(sayt.hideSayt).to.not.be.called;
+    });
+
+    it('should not hide SAYT if the event target is the provided search box', () => {
+      stub(sayt, 'contains').returns(false);
+      stub(sayt, 'nodeInSearchBar').returns(true);
+      stub(sayt, 'hideSayt');
+
+      sayt.processClick(event);
+
+      expect(sayt.contains).to.be.calledWith(node);
+      expect(sayt.nodeInSearchBar).to.be.calledWith(node);
+      expect(sayt.hideSayt).to.not.be.called;
+    });
+  });
+
+  describe('clickCloseSayt()', () => {
+    it('should close SAYT', () => {
+      const hideSayt = stub(sayt, 'hideSayt');
+      const event = { preventDefault: () => null };
+
+      sayt.clickCloseSayt(event);
+
+      expect(hideSayt).to.be.calledOnce;
+    });
+
+    it('should prevent the default event action', () => {
+      const preventDefault = spy();
+      const event = { preventDefault };
+      stub(sayt, 'hideSayt');
+
+      sayt.clickCloseSayt(event);
+
+      expect(preventDefault).to.be.calledOnce;
```

Comment:
Is it overtesting to assert that it was called exactly once? If I'm not mistaken, the result is the same if we call `preventDefault` more than once.

Reply:
Yes, this is definitely over-testing.